src/renderToString.js

```
@@ -0,0 +1,31 @@
+// @flow
+
+import Vue from 'vue'
+import createInstance from './lib/create-instance'
+import './lib/polyfills/object-assign-polyfill'
+import { throwError } from './lib/util'
+
+Vue.config.productionTip = false
+Vue.config.devtools = false
+
+export default function renderToString (component: Component, options: Options = {}): string {
+  const renderer = require('vue-server-renderer').createRenderer()
```

Is there a reason why `vue-server-renderer` gets imported using CommonJS and not ESM?

---

ESM imports are hoisted to the top, so any ESM import will always be bundled.

vue-server-renderer relies on node modules, like path, which makes it difficult to compile with webpack. require is only run when the code runs, so putting it inside the renderToString means we only try and import vue-server-renderer when the renderToString function is run.

We should also add a warning that checks if the user is in node, and we can throw an error if they aren't to tell them renderToString needs to be run in node.

---

Clear explanation, thank you very much!